_posts/2023-11-10-automated-test-generation-chess-puzzles-with-dafny.markdown

```diff
@@ -0,0 +1,288 @@
+---
+layout: post
+title:  "Automated Test Generation: Chess Puzzles with Dafny"
+date:   2023-11-10 11:00:00 -0500
+author: Sasha Fedchin
+---
+Given all the incredible examples that showcase how powerful Dafny is (did you know that part of Dafny is written in Dafny?), you might think that verification can fully replace testing. And in many cases it does! However if, like a number of industrial Dafny users, you want to integrate Dafny code with existing codebases, you will face challenges that verification alone might not solve:
+
+- Perhaps your code is actually a reference implementation for another program and you want to test the equivalence between the two.
+- You might be calling out to external libraries, which are not themselves verified, and would like to test them by calling your Dafny code.
```

Comment:
What's external? I think you mean code that's not written in Dafny.

Consider something like "Through its foreign function interface, Dafny allows implementing specifications using language other than Dafny. Dafny can not verify such external implementations. To still gain confidence in the correctness of such these, you can use Dafny to write tests on them."